engine/dlib/src/dlib/sslsocket.cpp

```diff
@@ -439,6 +440,7 @@ dmSocket::Result Receive(SSLSocket* socket, void* buffer, int length, int* recei
 
     if( ret <= 0 )
     {
+        mbedtls_ssl_session_reset(socket->m_SSLContext);
```

Comment:
From the mbedtls `mbedtls_ssl_read()`documentation:

<code>
If this function returns something other than a positive value,
 `#MBEDTLS_ERR_SSL_WANT_READ`, `#MBEDTLS_ERR_SSL_WANT_WRITE`, `#MBEDTLS_ERR_SSL_ASYNC_IN_PROGRESS`,
 `#MBEDTLS_ERR_SSL_CRYPTO_IN_PROGRESS` or `#MBEDTLS_ERR_SSL_CLIENT_RECONNECT`,
 you must stop using the SSL context for reading or writing,
 and either free it or call \c mbedtls_ssl_session_reset()
 on it before re-using it for a new connection; the current
 connection must be closed.
<code>

Since we call `mbedtls_ssl_close_notify()`on the context in dmSSLSocket::Delete(), I opted for using the `mbedtls_ssl_session_reset()`.

Reply:
Nice find!